### PR TITLE
Added DST Teacher for multiwoz_v22 task

### PR DESCRIPTION
**Pull request by prajjwal1**
The DST Teacher for `multiwoz_v22` is not present. Adding this teacher would allow `multiwoz_v22` to be usable for Dialog State Tracking task.

cc @chinnadhurai

### CODE REVIEW

File: parlai/tasks/multiwoz_v22/agents.py [hunk 1]
```diff
@@ -373,6 +379,212 @@ def get_id_task_prefix(self):
         return "MultiwozV22"
 
 
+class MultiWOZv22DSTTeacher(tod_agents.TodUserSimulatorTeacher):
```
Comment (chinnadhurai) [inline parlai/tasks/multiwoz_v22/agents.py:382]: might be better to add a comment saying this teacher is needed for reproducing the Joint Goal Accuracy values reported in the simpleTOD & SOLOIST papers.

Comment (prajjwal1) [reply parlai/tasks/multiwoz_v22/agents.py:382]: Is it better now ? @chinnadhurai

**Comment by moyapchen** [comment]
Looks okay code-wise, though I haven't done a through review. Should this be in ParlAI-Internal rather than public ParlAI though?

File: parlai/tasks/multiwoz_v22/agents.py [hunk 2]
```diff
@@ -393,5 +605,5 @@ class SingleApiSchemaAgent(MultiwozV22Parser, tod_agents.TodSingleApiSchemaAgent
     pass
 
 
-class DefaultTeacher(SystemTeacher):
```
Comment (chinnadhurai) [inline parlai/tasks/multiwoz_v22/agents.py:396]: would be better to keep the default to SystemTeacher as is.

**Comment by chinnadhurai** [comment]
> Looks okay code-wise, though I haven't done a through review. Should this be in ParlAI-Internal rather than public ParlAI though?

we need this teacher for the actual DST task in Multiwoz and reproduce JGA reported in the SOTA models. So, figured it might make sense to push to the public one.

File: parlai/tasks/multiwoz_v22/agents.py (hunk 2 above)
Comment (prajjwal1) [reply parlai/tasks/multiwoz_v22/agents.py]: Fixed this.

**Comment by prajjwal1** [comment]
@stephenroller Can you please look into this ? It has been here for sometime now.

**Review by stephenroller — COMMENTED** [review]
I'll defer to @moyapchen ans @klshuster here. Don't see any issue though

**Review by klshuster — COMMENTED** [review]
deferring to @moyapchen for final say but looks OK to me

could we [generate test files](http://parl.ai.s3-website.us-east-2.amazonaws.com/docs/tutorial_task.html#part-5-contributing-upstream) for it as well?

**Comment by prajjwal1** [comment]
@moyapchen Can you please see this ?

**Review by moyapchen — APPROVED** [review]
lgtm

**Pull request merged**